src/lib.rs

```diff
@@ -475,23 +479,33 @@ macro_rules! viewable_primitive {
     )*
   };
 }
+macro_rules! viewable_number_primitive {
+  ($($child_type:ty),* $(,)?) => {
+    $(
+      impl CellValue for $child_type {
+        #[inline(always)]
+        fn render_value(self, options: &RenderOptions) -> impl IntoView {
+        if let Some(value) = options.precision.as_ref() {
+            view! {
+                <>{format!("{:.value$}", self)}</>
+            }
+        }
+        else {
+            view! {
+                <>{self.to_string()}</>
+            }
+        }
+        }
+      }
+    )*
+  };
+}
+viewable_number_primitive![
+    usize, u8, u16, u32, u64, u128, isize, i8, i16, i32, i64, i128, f32, f64,
```

Comment:
this has a regression of not supporting Decimal, we should implement this behind a feature flag instead, should we do this in this pr?

Reply:
Yes, let's do a feature flag 👍🏼